subprojects/testing-base/src/test/groovy/org/gradle/api/internal/tasks/testing/junit/result/JUnitXmlResultWriterSpec.groovy

```diff
@@ -35,7 +35,7 @@ import static org.hamcrest.CoreMatchers.equalTo
 class JUnitXmlResultWriterSpec extends Specification {
```

Comment:
❗ In addition to unit tests, I think this needs a few integration tests.  I'd want to see at least 3:
 - stderr can be omitted
 - stdout can be omitted
 - the defaults are the both are included
 
Chec out `org.gradle.testing.junit.JUnitIntegrationTest#suitesOutputIsVisible()` in the `testing-jvm` project for inspiration.

Reply:
Integration tests (or 1 test covering these 3 scenarios) are still missing.  You can add them to the `JUnitIntegrationTest` class I pointed out.  

See the `given` block in the `executesTestsInCorrectEnviroment()` test for how to write to the build script and alter `test { reports { junitXml {} } }` properties.  Look at the `assertStdout(Matcher)` method used in the expectations of the `suitesOutputIsVisible()` test and implemented in `JUnitTestClassExecutionResult.groovy` to see how to check the presence of nodes in the XML.